File: gamulti/app/src/main/java/com/maxe10/gamulti/MainActivity.java
package com.maxe10.gamulti;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity
{
  private MultiDrawView multiDrawView;
  private Button buttonReset;

  @Override
  protected void onCreate(Bundle savedInstanceState)
  {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    multiDrawView = (MultiDrawView) findViewById(R.id.multiDrawView);

    buttonReset = (Button) findViewById(R.id.buttonReset);
    buttonReset.setOnClickListener(new View.OnClickListener()
    {
      @Override
      public void onClick(View view)
      {
        multiDrawView.reset();
      }
    });
  }

  @Override
  public boolean onCreateOptionsMenu(Menu menu)
  {
    // Inflate the menu; this adds items to the action bar if it is present.
    getMenuInflater().inflate(R.menu.menu_main, menu);
    return true;
  }
}

File: LocalVentilationController/app/src/main/java/com/maxeoneo/localventilationcontroller/MainActivity.kt
package com.maxeoneo.localventilation

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.webkit.WebSettings
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.PreferenceManager

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val webView: WebView = findViewById(R.id.webview)
        val settings: WebSettings = webView.settings
        settings.useWideViewPort = true
        settings.loadWithOverviewMode = true
        settings.javaScriptEnabled = true
        settings.domStorageEnabled = true

        val sharedPref: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(this)
        val remoteUrl: String = sharedPref.getString("url", "http://192.168.178.46").toString()
        val localUrl = "file:///android_asset/setup-guide.html"

        webView.webViewClient = object : WebViewClient() {
            override fun onPageFinished(view: WebView, url: String) {
                // If the remote URL loads successfully, do nothing (content is already replaced)
            }

            override fun onReceivedError(
                view: WebView,
                request: android.webkit.WebResourceRequest,
                error: android.webkit.WebResourceError
            ) {
                // On error, show the local page
                view.loadUrl(localUrl)
            }
        }

        // Load local page first
        webView.loadUrl(localUrl)
        // Then try to load remote URL
        webView.loadUrl(remoteUrl)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_settings -> {
                val intent = Intent(this, SettingsActivity::class.java)
                startActivity(intent)
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}
